src/D_functions_and_procedures.sql

```diff
@@ -368,39 +368,55 @@ SELECT title FROM release WHERE media_id=47460;
 
 -------------------D9 SIMILAR MOVIES SEARCH-------------------------
 --------------------------------------------------------------------
-
 --FINAL FUNCTION SIMILAR MOVIES
 --THIS COMBINES OTHER FUNCTIONS WHICH ARE DECLARED LOWER IN THIS FILE
 CREATE OR REPLACE FUNCTION get_similar_movies(input_media_id INTEGER)
 RETURNS TABLE (
 	media_id INTEGER,
-	count INTEGER
+	count INTEGER,
+	title TEXT
 ) AS $$
 
 DECLARE
 	input_media RECORD;
 BEGIN
 
   --get title id and country of input_media_id
-	SELECT DISTINCT m.media_id, r.title, mpc.country_id INTO input_media
+	SELECT DISTINCT m.media_id, t.name, mpc.country_id INTO input_media
 	FROM media m
-	JOIN release r USING (media_id)
+	JOIN title t USING (media_id)
 	LEFT JOIN media_production_country mpc USING(media_id)
 	WHERE m.media_id = input_media_id;
 	
-	RETURN query
+	CREATE TEMPORARY TABLE search_result (
+		media_id INTEGER,
+		total_count INTEGER
+		) ON COMMIT DROP;
+	
+	
+	INSERT INTO search_result (media_id, total_count)
 	SELECT com_res.media_id, SUM(com_res.count)::INTEGER AS total_count
 	FROM (
 		SELECT *  FROM get_count_of_movies_with_same_actors(input_media_id) 
 		UNION ALL
-		SELECT *  FROM find_movies_from_the_same_country(input_media.country_id)
+		SELECT *  FROM get_movies_from_the_same_country(input_media.country_id)
+		UNION ALL
+		SELECT * FROM get_movies_with_same_genre(input_media_id)
 		UNION ALL
-		SELECT *  FROM find_movie_similar_titles(input_media.title,3)
+		SELECT *  FROM get_movies_with_similar_titles(input_media.name,3)
 		UNION ALL
-		SELECT *  FROM find_movies_with_similar_crew(input_media_id)
+		SELECT *  FROM get_movies_with_similar_crew(input_media_id)
 	) AS com_res
 	GROUP BY com_res.media_id
 	ORDER BY total_count DESC;
+	
+	RETURN query
+	SELECT DISTINCT sr.media_id, sr.total_count, t.name
+	FROM search_result sr
+	JOIN title t USING (media_id)
+	JOIN title_title_type ttt USING (title_id)
+	WHERE ttt.title_type_id = 8
```

Comment:
I dont think you should hardcode id values like this, especially in a function

Reply:
so you want it to say 'original'?